graphql/schema.graphql

```diff
@@ -1,57 +1,99 @@
-"""User role"""
-enum Role {
-  """Administrator role (department staff etc.)"""
-  ADMIN
-  """User role (professor, student etc.)"""
-  USER
-}
-
 scalar Date
 
 type Query {
-  """Get the current user"""
+  """
+  Get the current user
+  """
   me: User
-  """Find a user by their id"""
+  """
+  Find a user by their id
+  """
   findUserById(id: ID!): User
-  """Get Teaching Preference Survey"""
+  """
+  Get Teaching Preference Survey
+  """
   survey: TeachingPreferenceSurvey!
-  """Get a list of courses for a given term"""
+  """
+  Get a list of courses for a given term
+  """
   courses(term: Term): [CourseSection!]
   """
   Schedule for a given term. If year is given, returns the most recent schedule generated for that year.
   """
   schedule(year: Int): Schedule
+  """
+  Get all courses preferences
+  """
+  coursePreferences: [CoursePreference!]
+  """
+  Get the all users
+  """
+  allUsers: [User!]
 }
 
-"""Generated schedule for a year"""
+"""
+Generated schedule for a year
+"""
 type Schedule {
-  """ID of the schedule"""
+  """
+  ID of the schedule
+  """
   id: ID!
-  """Year for the schedule"""
+  """
+  Year for the schedule
+  """
   year: Int!
-  """When the schedule was generated"""
+  """
+  When the schedule was generated
+  """
   createdAt: Date!
-  """Scheduled courses"""
+  """
+  Scheduled courses
+  """
   courses(term: Term!): [CourseSection!]
 }
 
 type Mutation {
-  """Login into a user account using email and password"""
+  """
+  Login into a user account using email and password
+  """
   login(username: String!, password: String!): AuthPayload!
-  """Logout the currently logged in user"""
+  """
+  Logout the currently logged in user
+  """
   logout: AuthPayload!
-  """Register a new user account"""
+  """
+  Register a new user account
+  """
   createUser(username: String!): CreateUserMutationResult!
-  """Updates a user given the user id."""
+  """
+  Updates a user given the user id.
+  """
   updateUser(input: UpdateUserInput!): UpdateUserMutationResult
-  """Change the password of the currently logged in user"""
+  """
+  Change the password of the currently logged in user
+  """
   changeUserPassword(input: ChangeUserPasswordInput!): Response!
-  """Reset a users password."""
+  """
+  Reset a users password.
+  """
   resetPassword(id: ID!): ResetPasswordMutationResult!
-  """Teaching preferences"""
+  """
+  Teaching preferences
+  """
   createTeachingPreference(input: CreateTeachingPreferenceInput!): Response!
-  """Generate schedule"""
+  """
+  Generate schedule
+  """
   generateSchedule(input: GenerateScheduleInput!): Response!
+  """
+  Edit schedule
+  """
+  editSchedule(input: EditCourseInput): Response!
+  """
+  Plug and Play Compnay 3's and 4's Algorithms
+  """
+  company(input: CompanyInput): Response!
```

Comment:
I thought we wanted `CompanyInput` to be part of `generateSchedule`?
I'm fine with either but just want to make sure.

Reply:
Either way would work, but I think it makes more sense to have 1 mutation instead of 2

Reply:
I will make it a single mutation then